File: Java/Leetcode/Array/LengthLastWord.java
package com.brianprojects.Leetcode.Array;

public class LengthLastWord {
    // Space Complexity: O(1) | Time Complexity: O(n)
    public int lengthOfLastWord(String s){
        if(s == null){
            return 0;
        }
        int count = 0;
        for(var i = s.length()-1; i>=0; i--){
            if(Character.isSpaceChar(s.charAt(i)) && count ==0){
                continue;
            }
            if(Character.isSpaceChar(s.charAt(i)) && count !=0){
                break;
            }
            count++;
        }

        return count;
    }
}

File: Python/Leetcode_Python/Arrays/LengthLastWord.py
# Time Complexity: O(n) || Space Complexity: O(1)
def lengthOfLastWord(s: str) -> int:
    if not s:
        return -1

    count = 0
    last = len(s) - 1

    while last > 0 and s[last] == " ":
        last -= 1

    while last >= 0 and s[last] != " ":
        count += 1

    return count
